Modernize build and publish workflow to use uv native commands

kdcokenny opened the pull request:
## Summary

This PR modernizes our build and publishing workflow to align with uv's best practices, replacing the legacy `twine` approach with uv's native publishing commands.

## Changes

### 🔧 Build & Publishing Updates
- **Replace twine with `uv publish`** for both PyPI and TestPyPI
- **Add TestPyPI configuration** to `pyproject.toml` for easier testing
- **Enhanced package verification** using uv's recommended `--with` and `--no-project` flags
- **Remove twine dependency** from both `[project.optional-dependencies]` and `[tool.uv]` sections

### 📦 Version Management
Added new Makefile commands for version management:
- `make version` - Show current version
- `make version-patch` - Bump patch version (e.g., 1.0.0 → 1.0.1)
- `make version-minor` - Bump minor version (e.g., 1.0.0 → 1.1.0)
- `make version-major` - Bump major version (e.g., 1.0.0 → 2.0.0)

### 🔍 Improved Package Verification
The `make verify` command now:
- Uses `uv run --with postcrawl --no-project` to test installation
- Includes `--refresh-package` flag to bypass cache
- Tests both basic import and client import

## Benefits

1. **Simpler workflow** - No need for separate `twine` installation
2. **Better integration** - Native uv commands work seamlessly with other uv features
3. **Improved testing** - Package verification now matches uv's recommended practices
4. **Version management** - Easy version bumping without manual edits

## Testing

Before merging, please test:
```bash
# Build the package
make build

# Verify it installs correctly
make verify

# Test publishing to TestPyPI (requires TestPyPI token)
export UV_PUBLISH_TOKEN="your-testpypi-token"
make publish-test
```

## Migration Notes

When publishing, use environment variables for authentication:
```bash
# For PyPI
export UV_PUBLISH_TOKEN="your-pypi-api-token"

# For TestPyPI  
export UV_PUBLISH_TOKEN="your-testpypi-api-token"
```

The old `twine` commands are no longer needed.

kdcokenny committed:
modernize: Update build and publish workflow to use uv native commands

- Replace twine with uv publish for both PyPI and TestPyPI
- Add TestPyPI configuration to pyproject.toml
- Enhance package verification using uv's recommended approach
- Add version management commands (bump patch/minor/major)
- Remove twine dependency as it's no longer needed

This brings the project in line with uv's best practices for building
and publishing Python packages.

ps428 commented on Makefile:
```diff
@@ -116,12 +123,30 @@ build: clean
 
 # Verify package installation
 verify: build
-	uv run python verify_package.py
```
please remove the verify_pacakge.py now. deprecated post introduction of uv native commands.

kdcokenny replied:
done

kdcokenny committed:
chore: Remove deprecated verify_package.py

This file is no longer needed after updating to uv's native
verification approach using 'uv run --with' command in the Makefile.

ps428 reviewed (approved):
all good
